### PR TITLE
Python: Document internal AST classes.

### PR DESCRIPTION
**Pull request by tausbn**
We already document these in the classes that override them, so I simply added a pointer to this information.
I am creating a separate internal PR for the changes to the code that generates `AstGenerated.qll`. This PR should be safe to merge in isolation.

### CODE REVIEW

File: python/ql/src/semmle/python/Stmts.qll [hunk 1]
```diff
@@ -405,22 +407,25 @@ class TemplateWrite extends TemplateWrite_ {
...
 
 /** A list of statements */
 class StmtList extends StmtList_ {
-    /** Whether this list of statements contains s */
+    /** Holds if this list of statements contains the AST node `a` */
     predicate contains(AstNode a) {
         exists(Stmt item | item = this.getAnItem() | item = a or item.contains(a))
     }
 
+    /** Gets the last item in this list of statements. */
```
Comment (yoff) [inline python/ql/src/semmle/python/Stmts.qll:429]:
```suggestion
    /** Gets the last item in this list of statements, if any such exists. */
```

File: python/ql/src/semmle/python/Stmts.qll [hunk 2]
```diff
@@ -90,8 +90,10 @@ class AugAssign extends AugAssign_ {
     /* syntax: Expr += Expr */
     override Expr getASubExpression() { result = this.getOperation() }
 
+    /** Gets the target of this augmented assignment statement. */
```
Comment (yoff) [inline python/ql/src/semmle/python/Stmts.qll:93]: I assume that would be `x` in `x += 1`?

File: python/ql/src/semmle/python/Stmts.qll [hunk 3]
```diff
@@ -90,8 +90,10 @@ class AugAssign extends AugAssign_ {
...
     Expr getTarget() { result = this.getOperation().(BinaryExpr).getLeft() }
 
+    /** Gets the value of this augmented assignment statement. */
```
Comment (yoff) [inline python/ql/src/semmle/python/Stmts.qll:96]: Would that be `1` in `x += 1`? That is the RHS but not the final value of `x` so potentially a bit confusing. Thus I think the comment should spell it out.

File: python/ql/src/semmle/python/Stmts.qll (hunk 1 above)
Comment (tausbn) [reply python/ql/src/semmle/python/Stmts.qll:429]: [Style guide](https://github.com/github/codeql/blob/master/docs/qldoc-style-guide.md#predicates-with-result) says a simple "if any" is preferred, so I will do that.

File: python/ql/src/semmle/python/Stmts.qll (hunk 2 above)
Comment (tausbn) [reply python/ql/src/semmle/python/Stmts.qll]: Yep. I've updated the comment to say this explicitly.

File: python/ql/src/semmle/python/Stmts.qll (hunk 3 above)
Comment (tausbn) [reply python/ql/src/semmle/python/Stmts.qll]: Agreed. I've updated the comment.

**Comment by tausbn** [comment]
Thank you for the review! For the internal changes, see https://git.semmle.com/Semmle/code/pull/37200 :slightly_smiling_face:
(Not a priority -- just needs to be merged before we mess with the AST again.)

File: python/ql/src/semmle/python/Stmts.qll (hunk 1 above)
Comment (yoff) [reply python/ql/src/semmle/python/Stmts.qll]: True, that is fine.

**Comment by RasmusWL** [comment]
Hmm, I'm a bit conflicted on the addition of INTERNAL to all the classes of `python/ql/src/semmle/python/AstGenerated.qll`. I get that you're not supposed to be using the classes explicitly, but you _are_ supposed to use the member predicates. I don't really have a proposal for a better solution though :neutral_face:

**Comment by tausbn** [comment]
I get your point, but I think this use is fine? You won't see any indications of internal-ness when calling the member predicate, as it itself is not marked as internal. You'll see it is if you're, say, trying to extend one of the internal classes (and happen to activate the IntelliSense), but in this case, you very much want to know that you should be extending the external class instead. (And at any rate you could only do it in a `.qll` file, as the autogenerated classes all have the `library` annotation.)

In the long run, it might be better to simply reimplement all of the methods on the external classes, and turn the generated part into an inaccessible `private` module.

**Pull request merged**